src/transformers/models/funnel/modeling_tf_funnel.py

```diff
@@ -144,7 +144,7 @@ def init_attention_inputs(self, inputs_embeds, attention_mask=None, token_type_i
         # attention_mask and token_type_ids have shape batch_size x seq_len
         self.pooling_mult = 1
         self.seq_len = seq_len = shape_list(inputs_embeds)[1]
-        position_embeds = self.get_position_embeds(seq_len, dtype=inputs_embeds.dtype, training=training)
```

Comment:
why remove the `type` hint here? How does `position_embeds` knows whether it should be float16 or float32?

Reply:
It will takes the `dtype` of the weights of the `position_embeddings` which will have the proper `dtype`.